### PR TITLE
Create Wheres

### PR DESCRIPTION
**Pull request by k-stewart**
First part of https://github.com/EFForg/eff_fab/issues/90: create the Where model.

### CODE REVIEW

File: db/migrate/20180312205753_create_wheres.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateWheres < ActiveRecord::Migration
+  def change
+    create_table :wheres do |t|
+      t.integer :user_id
+      t.datetime :sent_at
+      t.text :body
+      t.string :provenance
```
Comment (wioux) [inline db/migrate/20180312205753_create_wheres.rb]: What's this one for?

File: app/models/where.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Where < ActiveRecord::Base
+  belongs_to :user
+
+  validates_presence_of :user_id, :body
+
+  before_create :ensure_sent_at
+
+  def ensure_sent_at
+    sent_at = Time.now if sent_at.blank?
```
Comment (wioux) [inline app/models/where.rb]: There's a missing `self.` here.

File: db/migrate/20180312205753_create_wheres.rb (hunk 1 above)
Comment (k-stewart) [reply db/migrate/20180312205753_create_wheres.rb]: For recording whether the status came from email or Mattermost

**Comment by k-stewart** [comment]
@wioux what do you think about this name?  I am worried that it will confuse developers and computers alike.  Would it make sense to call the table `whereabouts`?

**Comment by wioux** [comment]
Yeah `wheres` isn't great for readability. Whereabouts is what we're getting at, but unfortunately it's one of those "defective nouns" with no plural, which I prefer to avoid. WhereMessage?

**Comment by k-stewart** [comment]
Works for me!

**Pull request merged**